controllers/openstackdataplanedeployment_controller.go

```diff
@@ -357,6 +367,9 @@ func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context,
 	Log.Info("Set DeploymentReadyCondition true")
 	instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
 	instance.Status.Deployed = true
+	if version != nil {
+		instance.Status.DeployedVersion = version.Spec.TargetVersion
+	}
```

Comment:
There will always be an OpenStackVersion resource in the namespace of the OpenStackControlplane. While it is okay to check for nil here, it might be better to have it wait for that resource to exist, and re-queue if it doesn't

Reply:
+1